ReactAndroid/src/main/java/com/facebook/react/animated/NativeAnimatedNodesManager.java

```diff
@@ -363,12 +363,22 @@ public void removeAnimatedEventFromView(int viewTag, String eventName, int anima
   }
 
   @Override
-  public void onEventDispatch(Event event) {
-    // Only support events dispatched from the UI thread.
-    if (!UiThreadUtil.isOnUiThread()) {
-      return;
+  public void onEventDispatch(final Event event) {
+    // Events can be dispatched from any thread so we have to make sure handleEvent is run from the
+    // UI thread.
+    if (UiThreadUtil.isOnUiThread()) {
```

Comment:
On Android we have to explicitly check the thread to avoid dispatching if we are already on the main thread while on iOS this is already handled in RCTExecuteOnMainQueue